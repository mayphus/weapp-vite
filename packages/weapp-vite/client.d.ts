/// <reference types="vite/client" />

type MP_PLATFORM = 'weapp' | 'alipay' | 'tt'

interface ImportMetaEnv {
  MP_PLATFORM: MP_PLATFORM // | 'swan' | 'alipay' | 'tt' | 'qq' | 'jd' | 'h5'
}
// MP_PLATFORM
// weapp / swan / alipay / tt / qq / jd / h5
declare module 'process' {
  global {
    namespace NodeJS {
      interface ProcessEnv extends Dict<string> {
        MP_PLATFORM: MP_PLATFORM
      }
    }
  }
}
