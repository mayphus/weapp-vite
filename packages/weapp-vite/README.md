<div align="center">
  <a href="https://vite.icebreaker.top">
    <img width="200" height="200" hspace="10" src="https://vite.icebreaker.top/logo.png" alt="vite logo" />
  </a>
  <h1>Weapp Vite</h1>
  <p>
    给小程序以现代化的开发体验
  </p>
  <img src="https://img.shields.io/node/v/weapp-vite" alt="node-current" />
  <img src="https://img.shields.io/npm/dependency-version/weapp-vite/peer/vite" alt="npm peer dependency version" />
  <img src="https://img.shields.io/github/v/release/weapp-vite/weapp-vite" alt="GitHub release" />
  <img src="https://img.shields.io/npm/l/weapp-vite" alt="licence" />
</div>

<p>&nbsp;</p>

## 使用文档地址: [vite.icebreaker.top](https://vite.icebreaker.top)

## Features

- 使用 Vite 构建，带来了 `typescript` / `scss` / `less` 等等的原生支持
- Vite 插件生态支持，也可以自定义编写插件，方便扩展

## Contribute

我们邀请你来贡献和帮助改进 `weapp-vite` 💚💚💚

以下有几个方式可以参与:

- 报告错误：如果您遇到任何错误或问题，请提`issue`并提供完善的错误信息和复现方式。
- 建议：有增强 `weapp-vite` 的想法吗？请提 `issue` 来分享您的建议。
- 文档：如果您对文档有更好的见解或者更棒的修辞方式，欢迎 `pr`。
- 代码：任何人的代码都不是完美的，我们欢迎你通过 `pr` 给代码提供更好的质量与活力。

## License

[MIT](./LICENSE)
