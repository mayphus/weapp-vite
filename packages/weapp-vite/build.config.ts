import { fileURLToPath } from 'node:url'
// import { resolve } from 'pathe'
import { defineBuildConfig } from 'unbuild'

const srcDir = fileURLToPath(new URL('src', import.meta.url))

const stubAlias = {
  '@': srcDir,
}

export default defineBuildConfig({
  entries: [
    {
      input: 'src/index',
    },
    {
      input: 'src/cli.ts',
    },
    {
      input: 'src/config.ts',
    },
    {
      input: 'src/json.ts',
    },
    {
      input: 'src/volar.ts',
    },
    {
      input: 'src/auto-import-components/resolvers/index.ts',
      name: 'auto-import-components/resolvers',
    },
  ],
  stubOptions: {
    jiti: {
      alias: stubAlias,
    },
  },
  // 'index': 'src/index.ts',
  // 'cli': 'src/cli.ts',
  // 'config': 'src/config.ts',
  // 'json': 'src/json.ts',
  // 'volar': 'src/volar.ts',
  // 'auto-import-components/resolvers': 'src/auto-import-components/resolvers/index.ts',
  declaration: true,
})
