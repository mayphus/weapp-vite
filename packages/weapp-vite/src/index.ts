import 'reflect-metadata'

export * from './config'
export * from './createContext'
