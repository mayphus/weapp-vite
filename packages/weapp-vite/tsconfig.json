{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "target": "ESNext",
    "baseUrl": ".",
    "module": "ESNext",
    "moduleResolution": "node",
    "paths": {
      "@/*": [
        "src/*"
      ]
    },
    "resolveJsonModule": true,
    "types": [
      "vitest/globals"
    ],
    "strict": true,
    "declaration": true,
    "outDir": "./dist",
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "skipLibCheck": true
  },
  "include": [
    "src",
    "test"
  ]
}
