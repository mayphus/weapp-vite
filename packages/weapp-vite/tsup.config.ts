import { defineConfig } from 'tsup'

export default defineConfig({
  entry: {
    index: 'src/index.ts',
    cli: 'src/cli.ts',
    config: 'src/config.ts',
  },
  dts: true,
  clean: true,
  format: ['cjs', 'esm'],
  shims: true,
  outExtension({ format }) {
    return {
      js: `.${format === 'esm' ? 'mjs' : 'cjs'}`,
    }
  },
  // external: ['vite'],
})
