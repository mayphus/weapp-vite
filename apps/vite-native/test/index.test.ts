import CI from 'ci-info'
import { omit } from 'es-toolkit'
import path from 'pathe'
import { describe, expect, it } from 'vitest'
import { createCompilerContext } from 'weapp-vite'

function removePaths(obj: Record<string, any>) {
  return omit(obj, ['path', 'jsonPath', 'sitemapJsonPath', 'themeJsonPath'])
}

describe.skipIf(CI.isCI)('scan', () => {
  it('compilerContext scan vite-native', async () => {
    const ctx = await createCompilerContext({
      cwd: path.resolve(import.meta.dirname, '..'),
    })

    await ctx.build()

    const packageBEntriesCount = 2
    expect(ctx.entriesSet.size).toMatchSnapshot('entriesSet')
    expect(ctx.entries.length).toMatchSnapshot('entries')
    expect(Array.from(ctx.entriesSet).map(x => path.relative(ctx.cwd, x))).toMatchSnapshot()
    expect(ctx.subPackageMeta.packageB).toBeDefined()
    expect(ctx.subPackageMeta.packageB.entries.length).toBe(packageBEntriesCount)
    expect(ctx.subPackageMeta.packageB.entriesSet.size).toBe(packageBEntriesCount)
    // expect(appEntry).toMatchSnapshot()
    if (ctx.appEntry) {
      expect(removePaths(ctx.appEntry)).toMatchSnapshot()
    }
    expect(ctx.getPagesSet()).toMatchSnapshot('getPagesSet')
    // expect(ctx.potentialComponentMap).toMatchSnapshot('potentialComponentMap')
    //  expect(ctx.wxmlComponentsMap).toMatchSnapshot('wxmlComponentsMap')

    // console.log(ctx.entries)
  })
})
