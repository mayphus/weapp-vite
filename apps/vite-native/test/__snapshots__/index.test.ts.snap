// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`scan > compilerContext scan vite-native > entries 1`] = `26`;

exports[`scan > compilerContext scan vite-native > entriesSet 1`] = `26`;

exports[`scan > compilerContext scan vite-native > getPagesSet 1`] = `
Set {
  "pages/index/index",
  "pages/index/test",
  "pages/test/test",
  "pages/button/button",
  "pages/button/skyline/button",
  "pages/LoveFromChina/index",
  "pages/LoveFromChina/LoveFromChina",
  "packageA/pages/cat",
  "packageA/pages/dog",
  "packageB/pages/apple",
  "packageB/pages/banana",
}
`;

exports[`scan > compilerContext scan vite-native 1`] = `
[
  "app.js",
  "components/navigation-bar/navigation-bar.ts",
  "pages/index/index.ts",
  "components/avatar/avatar.ts",
  "pages/index/test.ts",
  "pages/test/test.ts",
  "components/HiChina/HiChina.ts",
  "components/icebreaker/index.ts",
  "pages/button/button.ts",
  "pages/button/base/index.js",
  "pages/button/size/index.js",
  "pages/button/icon-btn/index.js",
  "pages/button/group-btn/index.js",
  "pages/button/block-btn/index.js",
  "pages/button/ghost-btn/index.js",
  "pages/button/shape/index.js",
  "pages/button/theme/index.js",
  "pages/button/disabled/index.js",
  "pages/button/skyline/button.ts",
  "pages/LoveFromChina/index.ts",
  "components/HelloWorld/index.ts",
  "pages/LoveFromChina/LoveFromChina.ts",
  "packageA/pages/cat.ts",
  "packageA/pages/dog.ts",
  "custom-tab-bar/index.ts",
  "app-bar/index.ts",
]
`;

exports[`scan > compilerContext scan vite-native 2`] = `
{
  "json": {
    "appBar": {},
    "componentFramework": "glass-easel",
    "lazyCodeLoading": "requiredComponents",
    "pages": [
      "pages/index/index",
      "pages/index/test",
      "pages/test/test",
      "pages/button/button",
      "pages/button/skyline/button",
      "pages/LoveFromChina/index",
      "pages/LoveFromChina/LoveFromChina",
    ],
    "rendererOptions": {
      "skyline": {
        "defaultContentBox": true,
        "defaultDisplayBlock": true,
        "disableABTest": true,
        "sdkVersionBegin": "3.4.3",
        "sdkVersionEnd": "15.255.255",
      },
    },
    "sitemapLocation": "sitemap.json",
    "subPackages": [
      {
        "name": "pack1",
        "pages": [
          "pages/cat",
          "pages/dog",
        ],
        "root": "packageA",
      },
      {
        "independent": true,
        "name": "pack2",
        "pages": [
          "pages/apple",
          "pages/banana",
        ],
        "root": "packageB",
      },
    ],
    "tabBar": {
      "backgroundColor": "#000000",
      "color": "#000000",
      "custom": true,
      "list": [
        {
          "pagePath": "pages/index/index",
          "text": "首页",
        },
        {
          "pagePath": "pages/index/test",
          "text": "测试",
        },
      ],
      "selectedColor": "#000000",
    },
    "themeLocation": "theme.json",
    "usingComponents": {
      "navigation-bar": "/components/navigation-bar/navigation-bar",
    },
    "window": {
      "navigationBarBackgroundColor": "#ffffff",
      "navigationBarTextStyle": "black",
      "navigationBarTitleText": "Weixin",
    },
  },
  "sitemapJson": {
    "desc": "关于本文件的更多信息，请参考文档 https://developers.weixin.qq.com/miniprogram/dev/framework/sitemap.html",
    "rules": [
      {
        "action": "allow",
        "page": "*",
      },
    ],
  },
  "themeJson": {
    "dark": {
      "navBgColor": "#191919",
      "navTxtStyle": "white",
    },
    "light": {
      "navBgColor": "#f6f6f6",
      "navTxtStyle": "black",
    },
  },
  "type": "app",
}
`;
